tests/unit/recommenders/utils/test_notebook_utils.py

```diff
@@ -1,38 +1,148 @@
+    )
 
 
 @pytest.mark.notebooks
-def test_is_jupyter(output_notebook, kernel_name):
+def test_is_jupyter(notebook_types, output_notebook, kernel_name):
     # Test on the terminal
     assert is_jupyter() is False
     assert is_databricks() is False
 
     # Test on Jupyter notebook
-    path = Path(__file__).absolute().parent.joinpath("test_notebook_utils.ipynb")
-    pm.execute_notebook(
-        path,
+    execute_notebook(
+        notebook_types,
         output_notebook,
         kernel_name=kernel_name,
     )
-    nb = sb.read_notebook(output_notebook)
-    df = nb.scraps.dataframe
-    result_is_jupyter = df.loc[df["name"] == "is_jupyter", "data"].values[0]
-    assert result_is_jupyter  # is True not allowed
-    result_is_databricks = df.loc[df["name"] == "is_databricks", "data"].values[0]
-    assert not result_is_databricks
+    results = read_notebook(output_notebook)
+
+    assert results["is_jupyter"]
+    assert not results["is_databricks"]
 
 
 @pytest.mark.spark
 @pytest.mark.notebooks
 @pytest.mark.skip(reason="TODO: Implement this")
 def test_is_databricks():
     pass
+
+
+@pytest.mark.notebooks
+def test_notebook_execution(notebook_programmatic, output_notebook, kernel_name):
+    execute_notebook(
+        notebook_programmatic,
+        output_notebook,
+        kernel_name=kernel_name,
+    )
+
+    results = read_notebook(output_notebook)
+    assert results["response1"] == 3
+    assert results["response2"] is True
+    assert results["response3"] == 7
+
+
+@pytest.mark.notebooks
+def test_notebook_execution_int(notebook_programmatic, output_notebook, kernel_name):
+    execute_notebook(
+        notebook_programmatic,
+        output_notebook,
+        kernel_name=kernel_name,
+        parameters=dict(a=6),
+    )
+
+    results = read_notebook(output_notebook)
+    assert results["response1"] == 8
+
+
+@pytest.mark.notebooks
+def test_notebook_execution_float(notebook_programmatic, output_notebook, kernel_name):
+    execute_notebook(
+        notebook_programmatic,
+        output_notebook,
+        kernel_name=kernel_name,
+        parameters=dict(a=1.5),
+    )
+
+    results = read_notebook(output_notebook)
+    assert results["response1"] == 3.5
+
+
+@pytest.mark.notebooks
+def test_notebook_execution_letter(notebook_programmatic, output_notebook, kernel_name):
+    execute_notebook(
+        notebook_programmatic,
+        output_notebook,
+        kernel_name=kernel_name,
+        parameters=dict(b="M"),
+    )
+
+    results = read_notebook(output_notebook)
+    assert results["response2"] is True
+
+
+@pytest.mark.notebooks
+def test_notebook_execution_other_letter(
+    notebook_programmatic, output_notebook, kernel_name
+):
+    execute_notebook(
+        notebook_programmatic,
+        output_notebook,
+        kernel_name=kernel_name,
+        parameters=dict(b="A"),
+    )
+
+    results = read_notebook(output_notebook)
+    assert results["response2"] == "A"
```

Comment:
Why the input "A"'s response2 is "A" while "M"'s response2 is True, not "M"? Could you put some comments to the tests so that one can understand better?
btw, any reason you didn't utilize `parametrize` for these series of tests for different input values?

Reply:
Fixed in the next commit